### PR TITLE
[WIP] Adding Constant and Variable

### PR DESCRIPTION
**Pull request by Arafatk**

### CODE REVIEW

File: lib/tensorflow/graph.rb [hunk 1]
```diff
@@ -66,18 +67,27 @@ def define_op(opName, name , input, device, attrs)
...
     raise ("Operation does not exist.") if !op
     opName = op.name   # This ensures that case-sensitivity does not become an issue
+    input ||= []
     raise ("Invalid number of inputs.") if op.input_arg.length != input.length
     inputs = []
-    input.each do |node|
-      inputs.push(node.definition.name)
+    (0..input.length - 1).each do |i|
+      begin
+        if op.input_arg[i].is_ref and input[i].ref
```
Comment (zverok) [inline lib/tensorflow/graph.rb]: Never use `and` in this context, only `&&`. `and` has unexpected precedence and is not a logical operator, it is code flow operators (like `do_something and return`).

Comment (Arafatk) [inline lib/tensorflow/graph.rb]: @zverok Thanks

**Comment by Arafatk** [comment]
@zverok Sorry for the confusion, but I was trying various things to understand why circle ci is failing. I wanted to make a few changes to see if things work and then squash all the un necessary commits later. The current commit offers no changes to the gem but it is still failing by circle ci, so this needs to be examined.

**Comment by zverok** [comment]
Ah, ok, sorry to intercept in a random moment :)

**Comment by nethsix** [comment]
@Arafatk This branch has no circle.yml file to tell CircleCI how to perform CI, thus CircleCI tried to be smart and infer where the spec files are, and how to run the specs. Naturally this will fail because CircleCI did not build all the dependencies, e.g., Tensorflow, etc.

You can do any of the following:
1. Ignore this and proceed to merge if you have done your rspec testing locally
2. Merge the master (which already has the circle.yml) with this branch, and the re-push. Note: this probably won't run the CircleCI automatically because it is turned off for now since we were still testing how CircleCI can work best (see https://github.com/Arafatk/tensorflow.rb/pull/37)
3. Copy the circle.yml from master into this branch

Option 2 is probably best.

**Comment by nethsix** [comment]
@Arafatk I notice that the CI is failing. Let me try to fix it. This may have something to do with what @chrhansen mentioned to me about the branch name but didn't ring a bell back then.

**Pull request merged**